### PR TITLE
Add AcceptedAtActionResult tests

### PR DESCRIPTION
**Pull request by hughbe**
- Constructors
- UrlHelper
- OnFormatting

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs [hunk 1]
```diff
@@ -24,6 +22,60 @@ namespace Microsoft.AspNetCore.Mvc
 {
     public class AcceptedAtActionResultTests
     {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("actionName")]
+        public void Constructor_InitializesActionName(string actionName)
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs]: We generally don't test trivial property \ field assignments in our tests. Is there a reason you thought these were necessary?

File: test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs [hunk 2]
```diff
@@ -2,11 +2,9 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Buffers;
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs]: 👍

File: test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs [hunk 3]
```diff
@@ -135,6 +197,81 @@ public void Constructor_InitializesStatusCodeAndValue(object value)
...
+
+            // Act
+            var result = new AcceptedAtActionResult(actionName: null, controllerName: null, routeValues: null, value: null);
+
+            // Assert
+            Assert.Throws<ArgumentNullException>("provider", () => result.OnFormatting(context));
+        }
+
+        [Fact]
+        public void OnFormatting_NonNullUrlHelper_ReturnsUrlHelperAction()
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs]: Isn't this scenario covered by `ExecuteResultAsync_SetsObjectValueOfFormatter`?

File: test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs [hunk 4]
```diff
@@ -135,6 +197,81 @@ public void Constructor_InitializesStatusCodeAndValue(object value)
...
+            var result = new AcceptedAtActionResult("actionName", "controllerName", "routeValues", "value");
+
+            // Assert
+            Assert.Throws<ArgumentNullException>("context", () => result.OnFormatting(null));
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void OnFormatting_NonNullUrlHelperReturnsInvalidUrl_ThrowsInvalidOperationException(string url)
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs]: Isn't this covered by `ExecuteResultAsync_ThrowsIfActionUrlIsNull`?

**Comment by hughbe** [comment]
> Most of the scenarios seem atypical for our unit tests.

That was actually the reason - it seems that lots of trivial tests are missing from the unit tests. This is probably deliberate actually, and maybe generates more work than reward

File: test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs (hunk 3 above)
Comment (hughbe) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs]: Ah yes - I got confused by all the uber complicated mocking code. I'll remove this test then

File: test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs (hunk 4 above)
Comment (hughbe) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs]: Small difference: doesn't include `string.Empty`, but I'll modify the old test

**Comment by hughbe** [comment]
@pranavkm friendly ping - I've updated the PR

**Comment by pranavkm** [comment]
Sorry, I'm a bit caught up with a couple of other things. @Eilon could you have someone look at this?

**Comment by Eilon** [comment]
@kichalla can you do a final review & merge?

**Comment by kichalla** [comment]
Merged with commit: 1dc4b28bbeb37c7cdf0df60fd97cf88f6faa6a6e
Thanks for the PR!

**Pull request closed without merging**